plugins/BEdita/API/src/Controller/ObjectsController.php

```diff
@@ -308,6 +305,49 @@ public function resource($id)
         return null;
     }
 
+    /**
+     * Authorize resource.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The resource entity
+     * @return void
+     */
+    protected function authorizeResource(EntityInterface $entity): void
+    {
+        if ($this->request->is('delete')) {
+            $this->Authorization->authorize($entity, 'update');
+            $this->Authorization->authorize($entity, 'updateParents'); // can't delete object if some parent is protected
+
+            return;
+        }
+
+        if (!$this->request->is('patch')) {
+            return;
+        }
+
+        $this->Authorization->authorize($entity, 'update');
+        if ($this->Authorization->can($entity, 'updateParents')) {
+            return;
+        }
+
+        // locked by parent => check if trying to change uname or status
+        $data = (array)$this->request->getData();
+        foreach (['uname', 'status'] as $field) {
+            if (!Hash::check($data, $field)) {
+                continue;
+            }
+
+            if ($entity->get($field) !== Hash::get($data, $field)) {
+                throw new ForbiddenException(
+                    __d(
+                        'bedita',
+                        'Cannot change "{0}" field since object {1} is locked by parent',
+                        [$field, $entity->id]
+                    )
+                );
+            }
+        }
```

Comment:
Since `->can($entity, 'updateParents')` is a rather expensive check, I'd try to flip the logic in this method to actually run that check only if the request is trying to change a locked field (which is a rather fast check):

```suggestion
        $protectedFieldsChanged = array_filter(
            ['uname', 'status'],
            fn (string $field): bool => Hash::check($data, $field) && $entity->get($field) !== Hash::get($data, $field),
        );
        if (empty($protectedFieldsChanged) || $this->Authorization->can($entity, 'updateParents')) {
            return;
        }

        throw new ForbiddenException(
            __d( // Perhaps join all changed protected fields, although the message needs to be rewritten.
                'bedita',
                'Cannot change "{0}" field since object {1} is locked by parent',
                [$protectedFieldsChanged[0], $entity->id]
            )
        );
```